plugin/src/main/scala/chisel3/internal/plugin/BundleComponent.scala

```diff
@@ -96,7 +96,12 @@ private[plugin] class BundleComponent(val global: Global, arguments: ChiselPlugi
         case acc: ValDef if acc.symbol.isParamAccessor =>
           paramAccessors += acc.symbol
         case con: DefDef if con.symbol.isPrimaryConstructor =>
+          if (con.symbol.isPrivate) { // distinguish between package private and private
+            val msg = "Private bundle constructors cannot automatically be cloned"
+            global.reporter.error(con.pos, msg)
+          }
           primaryConstructor = Some(con)
+
```

Comment:
```suggestion
          } else {
            primaryConstructor = Some(con)
          }
```
By not setting the primary constructor, we prevent the unhelpful cascading error message, improving it from:
```
[error] /scratch/koenig/chisel3/src/main/scala/Test.scala:6:41: Private bundle constructors cannot automatically be cloned
[error] class MyRecord private (x: Int) extends Record {
[error]                                         ^
[error] ## Exception when compiling 84 sources to /scratch/koenig/chisel3/target/scala-2.13/classes
[error] scala.reflect.internal.Types$TypeError: constructor MyRecord in class MyRecord cannot be accessed in package <empty> from package <empty>
```
to
```
[error] /scratch/koenig/chisel3/src/main/scala/Test.scala:6:41: Private bundle constructors cannot automatically be cloned
[error] class MyRecord private (x: Int) extends Record {
[error]                                         ^
[error] /scratch/koenig/chisel3/src/main/scala/Test.scala:6:7: Unable to determine primary constructor!
[error] class MyRecord private (x: Int) extends Record {
[error]       ^
```
Perhaps we should also consider removing that second error entirely since it's not clear if it's useful, but this is at least much better.

Reply:
Would it be possible to include some hints on what to fix with the error message?
Like "Try making the constructor public such that Chisel can create copies of your Record"

Reply:
Yeah absolutely, I added a suggestion for that in a separate suggestion